rename_imports.py

```diff
@@ -0,0 +1,78 @@
+
+from LSP.plugin.core.file_watcher import FileWatcherEvent, FileWatcherProtocol, get_file_watcher_implementation, FileWatcher
+from LSP.plugin.core.typing import  List
+import os
+import sublime
+
+
+def register_rename_import_file_watcher(root_path: str):
+    file_watcher = get_file_watcher_implementation()
+    if file_watcher:
+        watcher = file_watcher.create(
+            root_path=root_path,
+            patterns=["**/*.ts", "**/*.tsx", "**/*.js", "**/*.jsx"],
+            events=["create", "delete"],
+            ignores=['**/node_modules/**'],
+            handler=HandleRenameImport
+        )
+
+
+queued_events = []  # type: List[FileWatcherEvent]
+
+
+class HandleRenameImport(FileWatcherProtocol):
+    def on_file_event_async(events: List[FileWatcherEvent]) -> None:
+        global queued_events
+        queued_events.append(events[0])
+        sublime.set_timeout(check_if_there_was_a_rename, 10)
+
+
+def check_if_there_was_a_rename():
+    global queued_events
+    if len(queued_events) == 2:
+        first_event, new_name = queued_events[0]
+        second_event, old_name = queued_events[1]
+        if (first_event == 'create' and second_event == 'delete'):
```

Comment:
I'm afraid that there is not really a reliable way to detect renames. It could be just two unrelated files being added and removed (for example when changing branches).

Reply:
I thought that chokidar has a way to detect a rename, but that is not the case.
there is a different lib, that supports rename detection => https://github.com/paulmillr/chokidar/issues/303#issuecomment-903710306

For the time being, I would rely on this logic:
```
        if len(self.events) == 2:
            create_event = next((event for event in self.events if event[0] == 'create'), None)
            delete_event = next((event for event in self.events if event[0] == 'delete'), None)
            if (create_event and delete_event):
                # Rename detected
                new_name = create_event[1]
                old_name = delete_event[1]
                self.prompt_rename(old_name, new_name)
```
If we have 2 events, 
and when we have a create and delete event,
only then prompt the rename....

>  It could be just two unrelated files being added and removed (for example when changing branches).

While than can happen, 
I do not know how to better to handle that than at the moment.

Reply:
> I thought that chokidar has a way to detect a rename, but that is not the case.
> there is a different lib, that supports rename detection => [paulmillr/chokidar#303 (comment)](https://github.com/paulmillr/chokidar/issues/303#issuecomment-903710306)

Interesting watcher. At least if we are to trust the author's claims.

I wouldn't want to switch to using it as that's too much work and the claims would have to be tested in practice also but we could take inspiration from it and implement support for rename events in our watcher.

> I do not know how to better to handle that than at the moment.

Me neither that's why I'm suggesting to remove the potentially dangerous option to rename without confirmation.

Reply:
> there is a different lib, that supports rename detection => [paulmillr/chokidar#303 (comment)](https://github.com/paulmillr/chokidar/issues/303#issuecomment-903710306)

BTW. The code style of that project is... interesting.
 - Blank lines between every line of code
 - excessive use of spaces (leading/trailing spaces within parenthesis for example)

https://github.com/fabiospampinato/watcher/blob/master/src/watcher.ts

It would not be fun to contribute to such project :)